### DISCUSSION TITLE
gameloop question

### DISCUSSION THREAD

**Discussion by martinziebiker** [hajimehoshi/ebiten#1685] [Q&A]
Hi, I like Ebiten. Is it possible to run this form of gameloop:

for {
start := time.Now()
update()
draw()
time.Sleep(20*time.Millisecond - time.Since(start))
} 

So that FPS == TPS but with cap? 
Maybe let draw() be empty and call myDraw() inside update(). But how to get access to (screen *ebiten.Image) like in Draw() ?

Will it work?

Thank You for advice.

**Comment by hajimehoshi** [comment]
Hi.

If you want to make FPS == TPS, put everything at `Draw` and leave `Update` empty. The problem is that the speed of the game depends on the environment like the display's refresh rate, so I don't recommend to make FPS == TPS.

**Reply by martinziebiker** [reply]
Thank You for reply. Putting everything in draw is truly not optimal solution. I was looking for solution more like

FPS == TPS
SetMaxTPS(60)
SetVsyncEnabled(false)
Displays refresh rate irrelevent

That's why I use simpliefied game loop example above. 
No problem. I will write my own game engine :-)

**Reply by hajimehoshi** [reply]
Another option is
1. Create an offscreen image
2. Render everything on the offscreen image at `Update`
3. Draw the offscreen onto the screen at `Draw`

so that you can get 60 TPS at every environment.

**Reply by hajimehoshi** [reply]
> No problem. I will write my own game engine :-)

Cheers :-)
